tests/test_zarr_encryption_ipfs.py

```
@@ -0,0 +1,166 @@
+    )
+
+    # Test with no encryption key
+
+    encrypted_no_transformer = HAMT(
+        store=IPFSStore(),
+        root_node_id=hamt1_root,
+        read_only=True,
+    )
+    loaded_failure = xr.open_zarr(store=encrypted_no_transformer)
+    # Accessing data should raise an exception since we don't have the encryption key or the transformer
+    with pytest.raises(Exception):
```

Ruff (0.8.2)

`pytest.raises(Exception)` should be considered evil

(B017)